substrate/client/offchain/src/lib.rs

```diff
@@ -160,7 +160,7 @@ impl<RA, Block: traits::Block, Storage> OffchainWorkers<RA, Block, Storage> {
 				"offchain-worker".into(),
 				num_cpus::get(),
 			)),
-			shared_http_client: api::SharedClient::new(),
+			shared_http_client: api::SharedClient::new().unwrap(),
```

Comment:
Ok, this requires unwrap anyway...

@bkchr Is it ok to move the `shared_http_client` to the `OffchainWorkerOptions type` such that we can bail early from the service builder if this fails?

Reply:
Oh, I thought that all the `unwrap()` calls were inside the test codes but this was not 🙃

Reply:
I think we should handle the error [here](https://github.com/paritytech/polkadot-sdk/blob/master/substrate/bin/node/cli/src/service.rs#L785-#L806) and just pass SharedHttpClient in the OffChainWorkerOptions instead to avoid breaking this api.

but better to wait what @bkchr thinks is the best way :)

Reply:
Okay. I'll wait for 😃

Reply:
> I think we should handle the error [here](https://github.com/paritytech/polkadot-sdk/blob/master/substrate/bin/node/cli/src/service.rs#L785-#L806) and just pass SharedHttpClient in the OffChainWorkerOptions instead to avoid breaking this api.

We will break the API any way :see_no_evil: As we need to do it, I think returning `Result` from `new` sounds better to me

Reply:
> > I think we should handle the error [here](https://github.com/paritytech/polkadot-sdk/blob/master/substrate/bin/node/cli/src/service.rs#L785-#L806) and just pass SharedHttpClient in the OffChainWorkerOptions instead to avoid breaking this api.
> 
> We will break the API any way 🙈 As we need to do it, I think returning `Result` from `new` sounds better to me

Do you mean this `new` method, `OffchainWorkers::new`?